### ISSUE TITLE
Execution hangs when accessing a object member function in select clause

### ISSUE THREAD

**Issue by LahiruJayasekara** [ballerina-platform/ballerina-lang#13146]
**Description:**
```
forever {
        from inputStream
        select std.getMarks() as marksOfStudent
        => (TeacherOutput[] emp) {
            foreach var e in emp {
                outputStream.publish(e);
            }
        }
    }

type Student object {
   string abcd = "Lahiru";
    int marks = 45;

    function getMarks() returns int {
        return self.marks;
    }
};
```
This code hangs in execution infinitely.

**Suggested Assignees (optional):**
@mohanvive , @gimantha , @grainier

**Comment by anupama-pathirage** [comment]
@mohanvive Can you please look into this?

**Comment by gimantha** [comment]
This issue is fixed in  #13761
